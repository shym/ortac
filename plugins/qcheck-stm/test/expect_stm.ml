let () = print_endline "test with existing sut type:"
let () = Ortac_qcheck_stm.main "lib.mli" "good_init" "sut"
let () = print_newline ()
let () = print_endline "test with non-existing sut type:"
let () = Ortac_qcheck_stm.main "lib.mli" "good_init" "absent"
let () = print_newline ()
let () = print_endline "test with a wrong type for init function"
let () = Ortac_qcheck_stm.main "lib.mli" "bad_init" "sut"
let () = print_newline ()
let () = print_endline "test with an absent init function"
let () = Ortac_qcheck_stm.main "lib.mli" "absent" "sut"

