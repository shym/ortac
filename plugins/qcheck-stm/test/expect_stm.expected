test with existing sut type:
This will generate qcheck-stm tests using `sut' as principal type and `good_init' as init function.

test with non-existing sut type:
Can't find sut

test with a wrong type for init function
Can't find init function

test with an absent init function
Can't find init function
